### ISSUE TITLE
WER calculation is slow

### ISSUE THREAD

**Issue by djpg** [mozilla/DeepSpeech#1511]
- **Deepspeech**: 0.2.0a5
- **OS Platform and Distribution**: Ubuntu 16.04
- **TensorFlow version (use command below)**: 1.6.0
- **Python version**: 3.5.2
- **CUDA/cuDNN version**: 9.0/7.1
- **GPU model and memory**: 2 x gtx 1080 ti
- **Exact command to reproduce**: 
``` bash
$ DeepSpeech.py --train_files /home/enigma-admin/O-ASR-CLARA/data/train.csv --dev_files /home/enigma-admin/O-ASR-CLARA/data/dev.csv --test_files /home/enigma-admin/O-ASR-CLARA/data/test.csv --decoder_library_path /home/enigma-admin/ASR-CLARA/models/language/libctc_decoder_with_kenlm.so --lm_binary_path /home/enigma-admin/O-ASR-CLARA/models/language/clara_5gram.klm --lm_trie_path /home/enigma-admin/O-ASR-CLARA/models/language/trie --alphabet_config_path /home/enigma-admin/O-ASR-CLARA/models/language/alphabet.txt --train_batch_size 64 --dev_batch_size 64 --test_batch_size 64 --n_hidden 2048 --epoch 500 --checkpoint_dir /home/enigma-admin/DeepSpeech/models/session/ --summary_dir /home/enigma-admin/DeepSpeech/models/summary/ --summary_secs 1756 --export_dir /home/enigma-admin/DeepSpeech/models/modelo/ --validation_step 20 --display_step 20
```
I'm training my own model with my own data and I set --display_step 20, so each 20 epoch I get the wer. **The problem is that WER calculation is very slow, spend about 25 hours to calculate it** and only takes 40 minutes to train each epoch.

```bash
I Training epoch 19...
I Training of Epoch 19 - loss: 121.293536                                                                                                                                                                  
100% (439 of 439) |################################| Elapsed Time: 0:38:52 Time:  0:38:52

100% (439 of 439) |################################| Elapsed Time: 1 day, 0:59:52 Time: 1 day, 0:59:52
I Validating epoch 20...
I Validation of Epoch 20 - WER: 0.466893, loss: 116.05238342285156, mean edit distance: 0.225472
```

Is it possible to reduce the WER calculation time?

**Comment by kdavis-mozilla** [comment]
In 0.2.0a5 yes, unfortunately, they WER calculation is slow.

However, in master the WER calculation has been re-worked and parallelized and it much faster[[1](https://github.com/mozilla/DeepSpeech/blob/3091b30b0516ef308944ddaaf6cf722c13b270ac/evaluate.py)].

However, one should know the neural network architecture of master has changed relative to 0.2.0a5 and one can not use old models there. We are currently training a model to release with this new architecture.

**Comment by djpg** [comment]
When does deepspeech.py use evaluation.py to calculate WER during training process?

**Comment by kdavis-mozilla** [comment]
It's now split out in to a second process. So you can call evaluation.py independently from deepspeech.py

**Comment by reuben** [comment]
Note that parallel decoding hasn't yet landed on master. `evaluate.py` is still pretty fast, but its main advantage is using the inference graph rather than the training one.
